### PR TITLE
praxisdigital/moodle-opgaver#503 - Made backup/restore tasks only have 1 attempt

### PR DESCRIPTION
**Pull request by frederikmillingpytlick**
Also refactored javascript & improved the reflection of the task status inside the block

### CODE REVIEW

File: amd/src/app/block/element.js [hunk 1]
```diff
@@ -252,9 +252,14 @@ export default class BlockElement {
             checkbox?.classList?.remove('d-none');
         }
 
-        this.#element.querySelector('.no-items')?.remove();
+        this.#element.querySelector('.no-items').classList.add('d-none');
```
Comment (AraPraxis) [inline amd/src/app/block/element.js:255]: Since '?' was used before, are we sure it's always found, so we don't get an error in the console?

Comment (frederikmillingpytlick) [reply amd/src/app/block/element.js:255]: Yup, i made it so it's no longer conditional and instead has a conditional class `d-none`

**Comment by frederikmillingpytlick** [comment]
Still isn't released on moodle.org, so no need to update the release - we should bump version tho, I totally missed that :grin:

**Review by AraPraxis — APPROVED** [review]

**Review by AraPraxis — APPROVED** [review]

**Comment by frederikmillingpytlick** [comment]
Tested OK with @AraPraxis

**Pull request merged**